src/router.rs

```diff
@@ -36,12 +36,14 @@ impl<T: Send> HttpRoutes<T> {
     }
 
     /// Register a request handler for a path.
+    #[allow(clippy::map_entry)]
     pub fn add_route(
         &mut self,
+        method: Method,
         path: String,
         handler: Box<dyn EndpointHandler<T> + Sync + Send>,
     ) -> Result<(), RouteError> {
-        let full = format!("{}{}", self.prefix, path);
+        let full = format!("{}:{}{}", method.to_str(), self.prefix, path);
```

Comment:
Can you also update the documentation of the function to say that it adds a `handler` for the (method, path) tuple? I Would also add that it needs to be unique.